test/ifElse/ifElseStep.js

```diff
@@ -188,6 +217,93 @@ describe('If Else step', () => {
                 ],
             })
         })
+
+        describe('as an entity', () => {
+            it('should audit the entity', async () => {
+                //given
+                const uc = givenASimpleUseCaseWithRequest()
+                const anEntity = entity('anEntiy',{
+                    param1: field(Number)
+                })
+
+                const anInstantiatedEntity = new anEntity()
+                anInstantiatedEntity.param1 = 1                
+
+                //when
+                await uc.run(anInstantiatedEntity)
+                //then
+                assert.deepStrictEqual(uc.auditTrail, {
+                    type: 'use case',
+                    description: 'A use case',
+                    transactionId: uc._mainStep._auditTrail.transactionId,
+                    elapsedTime: uc._mainStep._auditTrail.elapsedTime,
+                    request: {
+                        param1: 1
+                    },
+                    return: { Ok: { return1: 1 } },
+                    steps: [
+                        {
+                            type: 'if else',
+                            description: 'A condition',
+                            returnIf: {
+                                description: 'If Step',
+                                elapsedTime: uc._auditTrail.steps[0].returnIf.elapsedTime,
+                                type: 'step',
+                                return: { Ok: true }
+                            },
+                            returnThen: {
+                                description: "Then Step",
+                                elapsedTime: uc._auditTrail.steps[0].returnThen.elapsedTime,
+                                return: { Ok: '' },
+                                type: "step"
+                            },
+                            elapsedTime: uc._auditTrail.steps[0].elapsedTime,
+                        },
+                    ],
+                })
+            })
+
+            it('should audit the entity inside the entity cicularly', async () => {
```

Comment:
why this test is on `ifElseStep` file? How entity type on request is related to if/else step?

Reply:
I followed the place that was testing usecase with request, but I agree with you, these tests should be reviewed

![image](https://user-images.githubusercontent.com/4419301/149638536-af668d19-56b5-4b24-9ec4-f8f355459d06.png)

Reply:
Changes made, there was a place with a better fit for these tests afterall, thanks for the review.